crates/router/src/connector/globepay/transformers.rs

```diff
@@ -100,21 +124,138 @@ impl<F, T>
             types::PaymentsResponseData,
         >,
     ) -> Result<Self, Self::Error> {
-        Ok(Self {
-            status: enums::AttemptStatus::from(item.response.status),
-            response: Ok(types::PaymentsResponseData::TransactionResponse {
-                resource_id: types::ResponseId::ConnectorTransactionId(item.response.id),
-                redirection_data: None,
-                mandate_reference: None,
-                connector_metadata: None,
-                network_txn_id: None,
-            }),
-            ..item.data
-        })
+        if item.response.return_code == GlobepayReturnCode::Success {
+            let globepay_metadata = GlobepayConnectorMetadata {
+                image_data_url: item
+                    .response
+                    .qrcode_img
+                    .ok_or(errors::ConnectorError::ResponseHandlingFailed)?,
+            };
+            let connector_metadata = Some(common_utils::ext_traits::Encode::<
+                GlobepayConnectorMetadata,
+            >::encode_to_value(&globepay_metadata))
+            .transpose()
+            .change_context(errors::ConnectorError::ResponseHandlingFailed)?;
+            let globepay_status = item
+                .response
+                .result_code
+                .ok_or(errors::ConnectorError::ResponseHandlingFailed)?;
+
+            Ok(Self {
+                status: enums::AttemptStatus::from(globepay_status),
+                response: Ok(types::PaymentsResponseData::TransactionResponse {
+                    resource_id: types::ResponseId::ConnectorTransactionId(
+                        item.response
+                            .order_id
+                            .ok_or(errors::ConnectorError::ResponseHandlingFailed)?,
+                    ),
+                    redirection_data: None,
+                    mandate_reference: None,
+                    connector_metadata,
+                    network_txn_id: None,
+                }),
+                ..item.data
+            })
+        } else {
+            Ok(Self {
+                status: enums::AttemptStatus::Failure, //As this connector gives 200 in failed scenarios . if return_code is not success status is mapped to failure. ref = "https://pay.globepay.co/docs/en/#api-QRCode-NewQRCode"
+                response: Err(get_error_response(
+                    item.response.return_code,
+                    item.response.return_msg,
+                    item.http_code,
+                )),
```

Comment:
```suggestion
                response: get_error_response(item),
```

Reply:
we can't pass `item` as `item.response` is different in different flows (PaymentResponse, PaymentSyncResponse)